contracts/SpecificGroupStrategy.sol

```diff
@@ -0,0 +1,378 @@
+
+    /**
+     * @notice Used when a strategy does not meet the validator group health requirements.
+     * @param group The group's address.
+     */
+    error StrategyNotEligible(address group);
+
+    /**
+     * @notice Used when attempting to withdraw from specific group strategy
+     * but group does not have enough CELO. Group either doesn't have enough stCELO
+     * or it is necessary to rebalance the group.
+     * @param group The group's address.
+     * @param expected The expected vote amount.
+     * @param real The real vote amount.
+     */
+    error GroupNotBalancedOrNotEnoughStCelo(address group, uint256 expected, uint256 real);
+
+    /**
+     * @notice Used when there isn't enough CELO voting for an account's strategy
+     * to fulfill a withdrawal.
+     * @param group The group's address.
+     */
+    error CantWithdrawAccordingToStrategy(address group);
+
+    /**
+     * @notice Used when attempting to deposit when there are not active groups
+     * to vote for.
+     */
+    error NoActiveGroups();
+
+    /**
+     * @notice Used when attempting to deposit when chosen strategy cannot receive any new votes.
+     * @param group The group's address.
+     */
+    error StrategyCannotReceiveVote(address group);
+
+    /**
+     * @notice Used when attempting to withdraw but there are no groups being
+     * voted for.
+     */
+    error NoGroups();
+
+    /**
+     * @notice Initialize the contract with registry and owner.
+     * @param _registry The address of the Celo registry.
+     * @param _owner The address of the contract owner.
+     * @param _manager The address of the Manager contract.
+     */
+    function initialize(
+        address _registry,
+        address _owner,
+        address _manager
+    ) external initializer {
+        _transferOwnership(_owner);
+        __UsingRegistry_init(_registry);
+        __Managed_init(_manager);
+    }
+
+    /**
+     * @notice Set this contract's dependencies in the StakedCelo system.
+     * @param _account The address of the Account contract.
+     * @param _groupHealth The address of the GroupHealth contract.
+     * @param _defaultStrategy The address of the DefaultStrategy contract.
+     */
+    function setDependencies(
+        address _account,
+        address _groupHealth,
+        address _defaultStrategy
+    ) external onlyOwner {
+        require(_account != address(0), "Account null");
+        require(_groupHealth != address(0), "GroupHealth null");
+        require(_defaultStrategy != address(0), "DefaultStrategy null");
+
+        account = IAccount(_account);
+        groupHealth = IGroupHealth(_groupHealth);
+        defaultStrategy = IDefaultStrategy(_defaultStrategy);
+    }
+
+    /**
+     * @notice Marks a group as specific group strategy for voting.
+     * @param group The address of the group to add to the set of specific group
+     * strategies.
+     */
+    function allowStrategy(address group) external onlyOwner {
```

Comment:
Should we add a note in NATSPEC that `groupHealth.updateGroupHealth()` should be executed before running this?